flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/dialect/JDBCDialects.java

```diff
@@ -46,10 +59,84 @@
 		return Optional.empty();
 	}
 
-	private static class DerbyDialect implements JDBCDialect {
+	private abstract static class AbstractDialect implements JDBCDialect {
+
+		@Override
+		public void validate(TableSchema schema) throws ValidationException {
+			for (int i = 0; i < schema.getFieldCount(); i++) {
+				DataType dt = schema.getFieldDataType(i).get();
+				String fieldName = schema.getFieldName(i).get();
+
+				// TODO: We can't convert VARBINARY(n) data type to
+				//  PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO in LegacyTypeInfoDataTypeConverter
+				//  when n is smaller than Integer.MAX_VALUE
+				if (unsupportedTypes().contains(dt.getLogicalType().getTypeRoot()) ||
+						(!(dt.getLogicalType() instanceof LegacyTypeInformationType) &&
+						(VARBINARY == dt.getLogicalType().getTypeRoot()
+							&& Integer.MAX_VALUE != ((VarBinaryType) dt.getLogicalType()).getLength()))) {
+					throw new ValidationException(
+							String.format("The dialect don't support type: %s.", dt.toString()));
```

Comment:
String.format("The %s dialect doesn't support type: %s.", dialectName, dt.toString())